### ISSUE TITLE
Force sending encrypted message if key from peer is known

### ISSUE THREAD

**Issue by csb0730** [deltachat/deltachat-android#235]
When encrypted communication has been active with peer (both clients handling autocrypt) and peer uses a non autocrypt MUA in the communication the encryption is deactivated until peer uses autocrypt MUA again.
So far ok because this is autocrypt communication behaviour (in my assumption). 

But, it should be possible to force sending encrypted message even in the case the peer sends last message unencrypted from another client (non autocrypt). The public key from peer user is still known.
The sending DC user should be sure to send an encrypted message if he intends to do that !

Maybe this can be seen as a feature request and can be achieved by a chat specific setting (force encryption) or by a special send button or function?!

**Comment by r10s** [comment]
> The public key from peer user is still known.

Yes, the sender knows the key and has no problem in encrypting the messages. But we cannot be sure that the receiver can access his private key and can decrypt the message.

As Delta Chat just has no UI for enabling/disabling encryption, we just pause encryption. But I think this is clear. Maybe there will be an option for forcing encryption in the future, however, for now, this is out of scope - esp. as this may require more knowledge of how encryption works by the user.

A social workaround might be to ask the user if he can read encrypted mails. If he then answers from an Autocrypt-capable client, we usually can again encrypt to him.

Btw. Autocrypt itself defines an UI for enabling/disabling encryption just as requested, see https://autocrypt.org/level1.html#provide-a-recommendation-for-message-encryption

**Comment by csb0730** [comment]
`Yes, the sender knows the key and has no problem in encrypting the messages. But we cannot be sure that the receiver can access his private key and can decrypt the message.`

This is true but if I intend to force encryption it's more important to encrypt than peer is able to decrypt at any time or each MUA he uses.

This problem comes up if a peer uses two different MUA in parallel (my case). For example a) an autocrypt capable MUA like DC and b) a standard MUA. At the end the peer user will only be able to read the encrypted message with an a) MUA. But especially this use case is intended in this situation.

I think it would be worth to implement this. Otherwise it isn't possible to send an encrypted message if really needed (!)

**Comment by Ampli-fier** [comment]
I agree with csb0730. Encrypted communication is a main feature and should always be possible.
Being forced to wait for another encrypted message is awkward :) 
I just faced this situation in a chat with a security-expert ;-)

**Comment by csb0730** [comment]
From technical point of view there shouldn't be any problem or big effort to implement this?

a) new option for chat "always encrypt" or
b) special send button?

Whatever...   ;-)

**Comment by r10s** [comment]
The newest release of Delta Chat (0.13.0) mitigates this issue a little bit: Replies to messages that are encrypted by the sender will be send encrypted, too.

**Comment by csb0730** [comment]
I do not understand why this should mitigate this?
Isn't that not the standard behaviour anyway?
